rpc_util.go

```diff
@@ -140,6 +141,7 @@ type callInfo struct {
 	failFast  bool
 	headerMD  metadata.MD
 	trailerMD metadata.MD
+	peer      peer.Peer
```

Comment:
I see your point. Yes, it makes sense to remove this copy from the codepath of the users that don't care about this value.